scripts/coverage.sh

```diff
@@ -1,136 +1,66 @@
+if ! command -v grcov >/dev/null 2>&1; then
+  echo "Error: grcov not found.  Try |cargo install grcov|"
   exit 1
 fi
 
-set -e
-cd "$(dirname "$0")/.."
-source ci/_
+# shellcheck source=ci/rust-version.sh
+source "$here/../ci/rust-version.sh" nightly
 
-cargo="$(readlink -f "./cargo")"
+# Clean up
+rm -rf "$here/../target" "$here/../coverage"
+find "$here/.." -type f -name '*.prof*' -exec rm {} +
+find "$here/.." -type f -name '*lcov*' -exec rm {} +
 
-: "${CI_COMMIT:=local}"
-reportName="lcov-${CI_COMMIT:0:9}"
+export RUSTFLAGS="-C instrument-coverage"
+export LLVM_PROFILE_FILE="default-%p-%m.profraw"
 
 if [[ -z $1 ]]; then
-  packages=(--lib --all --exclude solana-local-cluster)
+  PACKAGES=(--lib --all --exclude solana-local-cluster)
 else
-  packages=("$@")
-fi
-
-coverageFlags=()
-coverageFlags+=(-Zprofile)               # Enable coverage
-coverageFlags+=("-Aincomplete_features") # Suppress warnings due to frozen abi, which is harmless for it
-if [[ $(uname) != Darwin ]]; then        # macOS skipped due to https://github.com/rust-lang/rust/issues/63047
-  coverageFlags+=("-Clink-dead-code")    # Dead code should appear red in the report
-fi
-coverageFlags+=("-Ccodegen-units=1")     # Disable code generation parallelism which is unsupported under -Zprofile (see [rustc issue #51705]).
-coverageFlags+=("-Cinline-threshold=0")  # Disable inlining, which complicates control flow.
-coverageFlags+=("-Copt-level=0")
-coverageFlags+=("-Coverflow-checks=off") # Disable overflow checks, which create unnecessary branches.
-
-export RUSTFLAGS="${coverageFlags[*]} $RUSTFLAGS"
-export CARGO_INCREMENTAL=0
-export RUST_BACKTRACE=1
-export RUST_MIN_STACK=8388608
-
-echo "--- remove old coverage results"
-if [[ -d target/cov ]]; then
-  find target/cov -type f -name '*.gcda' -delete
+  PACKAGES=("$@")
 fi
-rm -rf target/cov/$reportName
-mkdir -p target/cov
-
-# Mark the base time for a clean room dir
-touch target/cov/before-test
-
-# Force rebuild of possibly-cached proc macro crates and build.rs because
-# we always want stable coverage for them
-# Don't support odd file names in our repo ever
-if [[ -n $CI || -z $1 ]]; then
-  # shellcheck disable=SC2046
-  touch \
-    $(git ls-files :**/build.rs) \
-    $(git grep -l "proc-macro.*true" :**/Cargo.toml | sed 's|Cargo.toml|src/lib.rs|')
-fi
-
-#shellcheck source=ci/common/limit-threads.sh
-source ci/common/limit-threads.sh
-
-_ "$cargo" nightly test --jobs "$JOBS" --target-dir target/cov --no-run "${packages[@]}"
 
 TEST_ARGS=(
-  --nocapture
   --skip shred::merkle::test::test_make_shreds_from_data::
   --skip shred::merkle::test::test_make_shreds_from_data_rand::
   --skip shred::merkle::test::test_recover_merkle_shreds::
 )
 
-# most verbose log level (trace) is enabled for all solana code to make log!
-# macro code green always
-if RUST_LOG=solana=trace _ ci/intercept.sh "$cargo" nightly test --jobs "$JOBS" --target-dir target/cov "${packages[@]}" -- "${TEST_ARGS[@]}" ; then
```

Comment:
can you restore this good parts from the arcane loving script maintained by me for years...? lol:

```bash
# most verbose log level (trace) is enabled for all solana code to make log!
# macro code green always
if RUST_LOG=solana=trace _ ci/intercept.sh ...
```

bonus: `RUST_LOG="solana=trace agave=trace $RUST_LOG"`.

Reply:
fixed: [aeb3e95](https://github.com/anza-xyz/agave/pull/1260/commits/aeb3e95880d65ace0768b5027765a2b20ebcce2a)